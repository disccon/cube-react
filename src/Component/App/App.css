body {
  font-family: "Open Sans", sans-serif;
  font-weight: 600;
  font-style: normal;
  overflow: visible;
}

.container {
  width: 100%;
  padding-top: 50px;
  padding-left: 50px;
}

.block-table {
  position: relative;
  display: inline-block;
  padding: 50px;
}
.block-table .wrapper-table {
  position: relative;
  padding: 2px;
}

/*# sourceMappingURL=App.css.map */
