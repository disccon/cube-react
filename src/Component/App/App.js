import React from 'react';
import TableCell from "../TableCell";

let setTimeoutMouseleave;
export default class App extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            initialWidth: 4,
            initialHeight: 4,
            cellSize: 50,
        };
        this.minusTop = React.createRef();
        this.minusLeft = React.createRef();
        this.plusBottom = React.createRef();
        this.plusRight = React.createRef();
        this.tableBody = React.createRef();
    }
    addRow = () => {
        if(1 === this.state.initialHeight) {
            this.minusTop.current.classList.add('button-minus_animation-display');
        }
        this.setState(prevState => ({
            initialHeight: prevState.initialHeight + 1
        }));
    }
    addCell = () => {
        if(1 === this.state.initialWidth) {
            this.minusLeft.current.classList.add('button-minus_animation-display');
        }
        this.setState(prevState => ({
            initialWidth: prevState.initialWidth + 1
        }));
    };
    deleteRow = () => {
        if(1 < this.state.initialWidth) {
            if(2 === this.state.initialWidth) {
                this.minusLeft.current.classList.remove('button-minus_animation-display');
            }
            this.setState(prevState => ({
                initialWidth: prevState.initialWidth - 1
            }));
            console.log(this.state.initialWidth)
        }
    }
    deleteCells = () => {
        if(1 < this.state.initialHeight) {
            if(2 === this.state.initialHeight) {
                this.minusTop.current.classList.remove('button-minus_animation-display');
            }
            this.setState(prevState => ({
                initialHeight: prevState.initialHeight - 1
            }));
        }
    }
    mouseOverTable = (event) => {
        let target = event.target
        clearTimeout(setTimeoutMouseleave)
        if (1 < this.state.initialWidth){
                this.minusLeft.current.classList.add('button-minus_animation-display');
        }
        if(1 < this.state.initialHeight){
                this.minusTop.current.classList.add('button-minus_animation-display');
        }
        if( target.tagName === "TD"){
            this.minusLeft.current.style.top = (target.parentElement.rowIndex * (this.state.cellSize +2) ) + 5 + "px";
            this.minusTop.current.style.left = (target.cellIndex * (this.state.cellSize + 2)) + 5+ "px";
        }
    }
    leaveTable = (event) => {
        setTimeoutMouseleave = setTimeout(() => {
            this.minusLeft.current.classList.remove('button-minus_animation-display');
            this.minusTop.current.classList.remove('button-minus_animation-display');
        }, 3000);
    }
    render() {
        return (
            <div className="block-table">
                <div className="wrapper-table" onMouseLeave={this.leaveTable} onMouseOver={this.mouseOverTable}>
                    <button type="button" className="button-minus button-minus_top" ref={this.minusTop} onClick={this.deleteCells}><span>-</span></button>
                    <button type="button" className="button-minus button-minus_left" ref={this.minusLeft} onClick={this.deleteRow}><span>-</span></button>
                    <button type="button" className="button-plus button-plus_right" ref={this.plusRight} onClick={this.addCell}>+</button>
                    <button type="button" className="button-plus button-plus_bottom" ref={this.plusBottom} onClick={this.addRow}>+</button>
                    <table className="table">
                        <TableCell widthRows={this.state.initialWidth} heightRows={this.state.initialHeight} sizeCell={this.state.cellSize}/>
                    </table>
                </div>
            </div>
        )
    }
}
