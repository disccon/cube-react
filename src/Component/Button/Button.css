.button-plus {
  position: absolute;
  z-index: 2;
  background: #f3a500;
  font-family: "Open Sans", sans-serif;
  font-weight: 600;
  font-size: 28px;
  line-height: 40px;
}

.button-plus, .button-minus {
  padding: 0px;
  font-family: "Open Sans", sans-serif;
  font-style: normal;
  color: #fff;
}

.button-plus:focus, .button-minus:active {
  box-shadow: 0 0 2px 3px rgba(72, 170, 230, 0.72);
}

.button-plus_right {
  top: 55px;
  right: 0px;
}

.button-plus_bottom {
  left: 55px;
}

.button-minus:hover, .button-plus:hover {
  transition: all ease 0.3s;
  opacity: 0.7;
}

.button-minus {
  display: flex;
  visibility: hidden;
  opacity: 0;
  justify-content: center;
  position: absolute;
  z-index: 2;
  background: #b20000;
  font-weight: 400;
  font-size: 37px;
  transition: opacity 0.4s ease-in-out, visibility 0.4s ease-in-out;
}
.button-minus span {
  height: 50px;
}

.button-minus_top {
  top: -50px;
  left: 5px;
}

.button-minus_left {
  top: 5px;
  left: -50px;
}

.button-minus_animation-display {
  visibility: visible;
  opacity: 1;
  transition: opacity 0.4s ease-in-out, visibility 0s ease-in-out, top 0.4s linear, left 0.4s linear;
}

/*# sourceMappingURL=Button.css.map */
