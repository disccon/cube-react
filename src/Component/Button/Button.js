import React from 'react';
import PropTypes from 'prop-types';

const Button = ({children,classButton,position,handleClick}) =>(
    <button type="button" className={classButton} style={position} onClick={handleClick}>
        {children}
    </button>
);

// Button.propTypes = {
//     cells: PropTypes.array.isRequired,
// };


export default Button;