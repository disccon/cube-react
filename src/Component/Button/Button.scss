.button-plus{
  position: absolute;
  z-index: 2;
  background: rgba(243, 165, 0, 1);
  font-family: 'Open Sans', sans-serif;
  font-weight: 600;
  font-size: 28px;
  line-height: 40px;
}
.button-plus,.button-minus{
  padding: 0px;
  font-family: 'Open Sans', sans-serif;
  font-style: normal;
  color: #fff;
}
.button-plus:focus,.button-minus:active{
  box-shadow: 0 0 2px 3px rgba(72,170,230,0.72);
}
.button-plus_right{
  top: 55px;
  right: -0px;
}
.button-plus_bottom{
  left: 55px;
}
.button-minus:hover,.button-plus:hover{
  transition: all ease .3s;
  opacity: 0.7;
}
.button-minus{
  display: flex;
  visibility: hidden;
  opacity: 0;
  justify-content: center;
  position: absolute;
  z-index: 2;
  background: rgba(178, 0, 0, 1);
  font-weight: 400;
  font-size: 37px;
  transition: opacity .4s ease-in-out, visibility .4s ease-in-out;
  span{
    height: 50px;
  }
}
.button-minus_top{
  top: -50px;
  left: 5px;
}
.button-minus_left{
  top: 5px;
  left: -50px;
}
.button-minus_animation-display{
  visibility: visible;
  opacity: 1;
  transition: opacity .4s ease-in-out, visibility 0s ease-in-out ,top .4s linear,left .4s linear;
}