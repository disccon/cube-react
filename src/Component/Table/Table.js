import React from 'react';
import PropTypes from 'prop-types';


const Table = ({children,sizeStyle}) => (
    <table className="table">
        <tbody>
            {children.map((trLists,index) => {
                return <tr key={trLists[0]}>
                        {trLists.map((tdList,index) => {
                            return <td key={tdList} style={sizeStyle}>{tdList}</td>
                        })
                        }
                </tr>
            })
            }
        </tbody>
    </table>
);
Table.propTypes = {
    children: PropTypes.array.isRequired,
    styleSize: PropTypes.object.isRequired,

};
Table.defaultProps = {
    styleSize:  {
        width: 50 + "px",
        height: 50 + "px"
    }
};
export default Table;