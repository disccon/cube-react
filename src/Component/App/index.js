import React from 'react'
import App from "./App";
import uuid from 'uuid/v4';
import {
    compose,
    withState,
    branch,
    withHandlers,
    renderComponent,
    mapProps,
    lifecycle,
} from 'recompose';



const  createTable = (props) => {
    let table = []
    let sizeStyle = {
        width: props.sizeCell + "px",
        height: props.sizeCell + "px"
    };
    for (let i = 0; i < props.heightRows ; i++) {
        let children = []
        for (let j = 0; j < props.widthRows; j++) {
            children.push(<td key={uuid()} style={sizeStyle}></td>)
        }
        table.push(<tr key={uuid()}>{children}</tr>)
    }
    return table
}

const withApp = compose(
    withState('initialWidth','setinitialWidth',4),
    withState('initialHeight','setinitialHeight',4),
    withState('cellSize','setcellSize',50),
    mapProps((props) => ({
        cells: createTable(props)
    })),
);

export default compose(withApp)(App);