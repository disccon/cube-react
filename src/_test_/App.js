import React from 'react';
//import { expect } from 'chai';
import App from '../Component/App/App';
import Button from "../Component/Button/Button";
import Table from "../Component/TableBody/Table";











// it('should render a App', () => {
//     const wrapper = mount(<App/>);
//     expect(wrapper.find(Table).get(0).props.children).toEqual([[11, 12, 13, 14], [21, 22, 23, 24], [31, 32, 33, 34], [41, 42, 43, 44]]);
//     wrapper.find('.button-plus_right').simulate('click');
//     wrapper.find('.button-plus_right').simulate('click');
//     //expect(wrapper).toMatchSnapshot();
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[11, 12, 13, 14, 15, 16], [21, 22, 23, 24, 25, 26], [31, 32, 33, 34, 35, 36], [41, 42, 43, 44, 45, 46]]);
//     wrapper.find('.button-plus_bottom').simulate('click');
//     wrapper.find('.button-plus_bottom').simulate('click');
//     //expect(wrapper).toMatchSnapshot();
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[11, 12, 13, 14, 15, 16], [21, 22, 23, 24, 25, 26], [31, 32, 33, 34, 35, 36], [41, 42, 43, 44, 45, 46],[51, 52, 53, 54, 55, 56],[61, 62, 63, 64, 65, 66]]);
//     wrapper.find('.button-minus_top').simulate('click');
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[12, 13, 14, 15, 16], [ 22, 23, 24, 25, 26], [32, 33, 34, 35, 36 ], [42, 43, 44, 45, 46],[52, 53, 54, 55, 56],[62, 63, 64, 65, 66]]);
//     wrapper.find('.button-minus_left').simulate('click');
//     wrapper.find('.button-minus_left').simulate('click');
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[32, 33, 34, 35, 36 ], [42, 43, 44, 45, 46],[52, 53, 54, 55, 56],[62, 63, 64, 65, 66]]);
// });


// it('Not work', () => {
//     expect(wrapper.find('.table').childAt(0).childAt(1).childAt(1).type()).toEqual('td');
//     expect(wrapper.state().minusTop).toEqual(5);
//     expect(wrapper.state().minusLeft).toEqual(5);
//     wrapper.find('.table').childAt(0).childAt(1).childAt(1).simulate('mouseover');
//     expect(wrapper.state().minusTop).toEqual(5); //toEqual() = 3  // minusTop: target.offsetLeft + 3, target.offsetLeft - not work
//     expect(wrapper.state().minusLeft).toEqual(5); //3  // minusLeft: target.offsetTop + 3, target.offsetTop - not work
// });