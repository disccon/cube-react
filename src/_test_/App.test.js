import React from 'react';
//import { expect } from 'chai';
import App from '../Component/App/App';
import Button from "../Component/Button/Button";
import Table from "../Component/Table/Table";

// test('should render App Jest ', () => {
//     const component = renderer.create(
//         <App/>,
//     );
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });


// it('should render App Mount', () => {
//     const wrapper = mount(
//         <App/>
//     );
//     expect(wrapper).toMatchSnapshot();
// });

// it('should render  App Shallow', () => {
//     const wrapper = shallow(
//         <App/>
//     );
//     expect(wrapper).toMatchSnapshot();
// });


// it('State App', () => {
//     const wrapper = mount(<App/>);
//     //expect(wrapper.find('.block-table').childAt(0)).toEqual('.wrapper-table');
//     expect(wrapper.state().tableCell).toEqual( [[11, 12, 13, 14], [21, 22, 23, 24], [31, 32, 33, 34], [41, 42, 43, 44]]);
//     expect(wrapper.state().initialWidth).toEqual(4);
//     expect(wrapper.state().initialHeight).toEqual(4);
//     expect(wrapper.state().cellSize).toEqual(50);
//     expect(wrapper.state().minusTopDisplay).toEqual(false);
//     expect(wrapper.state().minusLeftDisplay).toEqual(false);
//     expect(wrapper.state().minusTop).toEqual(5);
//     expect(wrapper.state().minusLeft).toEqual(5);
// });

// it('should render a App', () => {
//     const wrapper = mount(<App/>);
//     expect(wrapper.find('.wrapper-table').length).toBe(1)
//     expect(wrapper.find( 'Button')).toHaveLength(4);
//     expect(wrapper.find(Button).length).toBe(4)
//     expect(wrapper.find(Table).length).toBe(1)
//     expect(wrapper.find('.button-minus')).toHaveLength(2);
//     expect(wrapper.find('.button-minus').at(1).type()).toEqual('button')
//     expect(wrapper.find('.table').childAt(0).type()).toEqual('tbody');
//     expect(wrapper.find('.table').childAt(0).childAt(0).key()).toEqual("11");
//     expect(wrapper.find('.table').childAt(0).childAt(2).childAt(2).key()).toEqual("33")
// });



// it('Table get props', () => {
//     const wrapper = mount(<App/>);
//     expect(wrapper.find(Table).get(0).props.children).toEqual([[11, 12, 13, 14], [21, 22, 23, 24], [31, 32, 33, 34], [41, 42, 43, 44]]);
//     expect(wrapper.find(Table).get(0).props.sizeStyle).toEqual({width: "50px", height: "50px"});
// });


// it('shallow render Table', () => {
//     const wrapper = shallow(<Table sizeStyle={{width:  "50px", height:  "50px"}} children={[[11, 12, 13, 14], [21, 22, 23, 24], [31, 32, 33, 34], [41, 42, 43, 44]]}/>);
//     expect(wrapper).toMatchSnapshot();
// });


// it('should render a App', () => {
//     const wrapper = mount(<App/>);
//     expect(wrapper.find(Table).get(0).props.children).toEqual([[11, 12, 13, 14], [21, 22, 23, 24], [31, 32, 33, 34], [41, 42, 43, 44]]);
//     wrapper.find('.button-plus_right').simulate('click');
//     wrapper.find('.button-plus_right').simulate('click');
//     //expect(wrapper).toMatchSnapshot();
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[11, 12, 13, 14, 15, 16], [21, 22, 23, 24, 25, 26], [31, 32, 33, 34, 35, 36], [41, 42, 43, 44, 45, 46]]);
//     wrapper.find('.button-plus_bottom').simulate('click');
//     wrapper.find('.button-plus_bottom').simulate('click');
//     //expect(wrapper).toMatchSnapshot();
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[11, 12, 13, 14, 15, 16], [21, 22, 23, 24, 25, 26], [31, 32, 33, 34, 35, 36], [41, 42, 43, 44, 45, 46],[51, 52, 53, 54, 55, 56],[61, 62, 63, 64, 65, 66]]);
//     wrapper.find('.button-minus_top').simulate('click');
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[12, 13, 14, 15, 16], [ 22, 23, 24, 25, 26], [32, 33, 34, 35, 36 ], [42, 43, 44, 45, 46],[52, 53, 54, 55, 56],[62, 63, 64, 65, 66]]);
//     wrapper.find('.button-minus_left').simulate('click');
//     wrapper.find('.button-minus_left').simulate('click');
//     expect(wrapper.find('Table').get(0).props.children).toEqual([[32, 33, 34, 35, 36 ], [42, 43, 44, 45, 46],[52, 53, 54, 55, 56],[62, 63, 64, 65, 66]]);
// });


// it('Not work', () => {
//     expect(wrapper.find('.table').childAt(0).childAt(1).childAt(1).type()).toEqual('td');
//     expect(wrapper.state().minusTop).toEqual(5);
//     expect(wrapper.state().minusLeft).toEqual(5);
//     wrapper.find('.table').childAt(0).childAt(1).childAt(1).simulate('mouseover');
//     expect(wrapper.state().minusTop).toEqual(5); //toEqual() = 3  // minusTop: target.offsetLeft + 3, target.offsetLeft - not work
//     expect(wrapper.state().minusLeft).toEqual(5); //3  // minusLeft: target.offsetTop + 3, target.offsetTop - not work
// });