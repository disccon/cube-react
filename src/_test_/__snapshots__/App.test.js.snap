// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a App 1`] = `
<App>
  <div
    className="block-table"
  >
    <div
      className="wrapper-table"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <Button
        classButton="button-minus button-minus_top button-minus_animation-display"
        handleClick={[Function]}
        minusTop={3}
      >
        <button
          className="button-minus button-minus_top button-minus_animation-display"
          onClick={[Function]}
          style={
            Object {
              "left": "3px",
              "top": "undefinedpx",
            }
          }
          type="button"
        >
          <span>
            -
          </span>
        </button>
      </Button>
      <Button
        classButton="button-minus button-minus_left button-minus_animation-display"
        handleClick={[Function]}
        minusLeft={3}
      >
        <button
          className="button-minus button-minus_left button-minus_animation-display"
          onClick={[Function]}
          style={
            Object {
              "left": "undefinedpx",
              "top": "3px",
            }
          }
          type="button"
        >
          <span>
            -
          </span>
        </button>
      </Button>
      <Button
        classButton="button-plus button-plus_right"
        handleClick={[Function]}
      >
        <button
          className="button-plus button-plus_right"
          onClick={[Function]}
          style={
            Object {
              "left": "undefinedpx",
              "top": "undefinedpx",
            }
          }
          type="button"
        >
          <span>
            +
          </span>
        </button>
      </Button>
      <Button
        classButton="button-plus button-plus_bottom"
        handleClick={[Function]}
      >
        <button
          className="button-plus button-plus_bottom"
          onClick={[Function]}
          style={
            Object {
              "left": "undefinedpx",
              "top": "undefinedpx",
            }
          }
          type="button"
        >
          <span>
            +
          </span>
        </button>
      </Button>
      <Table
        sizeStyle={
          Object {
            "height": "50px",
            "width": "50px",
          }
        }
        styleSize={
          Object {
            "height": "50px",
            "width": "50px",
          }
        }
      >
        <div
          className="wrapper"
        >
          <table
            className="table"
          >
            <tbody>
              <tr
                key="11"
              >
                <td
                  key="11"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  11
                </td>
                <td
                  key="12"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  12
                </td>
                <td
                  key="13"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  13
                </td>
                <td
                  key="14"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  14
                </td>
              </tr>
              <tr
                key="21"
              >
                <td
                  key="21"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  21
                </td>
                <td
                  key="22"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  22
                </td>
                <td
                  key="23"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  23
                </td>
                <td
                  key="24"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  24
                </td>
              </tr>
              <tr
                key="31"
              >
                <td
                  key="31"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  31
                </td>
                <td
                  key="32"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  32
                </td>
                <td
                  key="33"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  33
                </td>
                <td
                  key="34"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  34
                </td>
              </tr>
              <tr
                key="41"
              >
                <td
                  key="41"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  41
                </td>
                <td
                  key="42"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  42
                </td>
                <td
                  key="43"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  43
                </td>
                <td
                  key="44"
                  style={
                    Object {
                      "height": "50px",
                      "width": "50px",
                    }
                  }
                >
                  44
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </Table>
    </div>
  </div>
</App>
`;
